### PR TITLE
feat: add FOX on arb as a related FOX asset

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box. This PR:

- Adds FOX on Arb as a *manual* related asset
- Merges manual and Zerion related assets in script (filtering out dupes) vs. having manual *or* zerion previously
- Regenerates asset data

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7075

Low to none

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- FOX on Arbitrum is available as a related FOX asset across the app, including in RFOx staking
- Do a quick smoke test on other related assets and ensure there's no regressions there

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

- Related assets dropdown is available in RFOX staking

Screenshot 2024-06-11 at 14 50 58

- Related assets are happy. This diff (left) vs. develop (right):

Screenshot 2024-06-11 at 14 50 34
Screenshot 2024-06-11 at 14 50 12
Screenshot 2024-06-11 at 14 49 54
Screenshot 2024-06-11 at 14 49 35

### CODE REVIEW

File: scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts [hunk 1]
```diff
@@ -230,25 +233,30 @@ const processRelatedAssetIds = async (
     })
   const manualRelatedAssetsResult = getManualRelatedAssetIds(assetId)
 
-  // ensure empty results get added so we can use this index to generate distinct asset list
```
Comment (gomesalexandre) [inline scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts:233]: Maybe one too deep in the diff side of things 💀

File: src/lib/asset-service/service/generatedAssetData.json [hunk 2]
```diff
@@ -158589,6 +158589,7 @@
     "symbol": "FOX",
     "explorer": "https://arbiscan.io",
     "explorerAddressLink": "https://arbiscan.io/address/",
-    "explorerTxLink": "https://arbiscan.io/tx/"
+    "explorerTxLink": "https://arbiscan.io/tx/",
+    "relatedAssetKey": "eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d"
```
Comment (gomesalexandre) [inline src/lib/asset-service/service/generatedAssetData.json:158593]: Mainnet FOX

File: scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts [hunk 3]
```diff
@@ -231,24 +234,30 @@ const processRelatedAssetIds = async (
...
-    relatedAssetsResult ?? {
-      relatedAssetIds: [],
-      relatedAssetKey: assetId,
-    }
+  const { relatedAssetIds: manualRelatedAssetIds } = manualRelatedAssetsResult ?? {
+    relatedAssetIds: [],
+  }
+
+  const relatedAssetKey =
+    manualRelatedAssetsResult?.relatedAssetKey || relatedAssetsResult?.relatedAssetKey || assetId
```
Comment (woodenfurniture) [inline scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts:242]: 🧠 explains the weth / eth related asset diff

File: scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts (hunk 1 above)
Comment (woodenfurniture) [reply scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts:233]: not bad at all

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**